### DISCUSSION TITLE
A Journey Into Shaders

### DISCUSSION THREAD

**Discussion by giscus** [SuperMayo/mayerowitz.io#3] [Announcements]
# A Journey Into Shaders

I finally decided to learn shaders. If you're curious about what shaders are and how they work, this article is for you. We'll start with the basics and build a simple blob from scratch.

https://www.mayerowitz.io/blog/a-journey-into-shaders

**Comment by pritster5** [comment]
Great guide!

**Reply by SuperMayo** [reply]
Thanks, hope it will be useful to others 🤞

**Comment by voidxd** [comment]
This is fantastic!

**Reply by SuperMayo** [reply]
Many thanks 🤗

**Comment by mrboen94** [comment]
This is actually so nice!
Well written and nice interactivity!

**Reply by SuperMayo** [reply]
Thanks a lot 🙏

**Comment by ram535** [comment]
Great article. May I ask which technology you used to write  this article?

**Reply by benpearson2003** [reply]
At a glance, I think three.js is the library.

**Reply by pindab0ter** [reply]
You can find the source code here: https://github.com/SuperMayo/mayerowitz.io/tree/4b8cbdc3a67ac5e4e1f61f5e286a212673a00dea/src/routes/blog/a-journey-into-shaders

**Reply by SuperMayo** [reply]
It's threeJS (I use a Svelte  wrapper called [Threlte](https://threlte.xyz/)). It's overkill for simple shaders, but I need it for future articles involving 3d geometries 👀
For the code editor I use [codemirror](https://codemirror.net/).
Finally the two svg animations are written in pure svg and controlled with a bit of javascript

**Comment by old-rob** [comment]
This article is excellent! Thank you!

**Reply by SuperMayo** [reply]
Thank you (for your comment and for the PR) 👍

**Comment by Achuttarsing** [comment]
Super clair ! Merci !

**Comment by cldprv** [comment]
Great guide

**Comment by Ennea** [comment]
I've finally gotten around to reading this blog post. I had been putting it off because I wanted to read it alongside my own journey into shaders.

I gotta say, this was excellent and has helped me tremendously! An amazing and very informative read, and the interactive examples were a nice touch, too.

Thank you very much, Antoine. I hope you will be writing more posts about your shader journey in the future :)
